app/controllers/api/v1/api_keys_controller.rb

```diff
@@ -47,19 +47,36 @@ def update
   private
 
   def check_mfa(user)
-    if user&.mfa_gem_signin_authorized?(otp)
+    if should_render_unauthorized_response?(user)
+      render_correct_unauthorized_response
+    elsif user&.mfa_gem_signin_authorized?(otp)
       return render_mfa_setup_required_error if user.mfa_required_not_yet_enabled?
       return render_mfa_strong_level_required_error if user.mfa_required_weak_level_enabled?
 
       yield
-    elsif user&.mfa_enabled?
-      prompt_text = otp.present? ? t(:otp_incorrect) : t(:otp_missing)
-      render plain: prompt_text, status: :unauthorized
     else
       false
     end
   end
 
+  def should_render_unauthorized_response?(user)
+    return false unless user
+    webauthn_enabled_no_otp_provided?(user) || mfa_enabled_no_otp_provided?(user)
+  end
+
+  def webauthn_enabled_no_otp_provided?(user)
+    user.webauthn_credentials.present? && otp.blank?
+  end
+
+  def mfa_enabled_no_otp_provided?(user)
+    !user.mfa_gem_signin_authorized?(otp) && user.mfa_enabled?
```

Comment:
`mfa_gem_signin_authorized?` checks if there is an MFA level set, then checks the webauthn otp / totp / recovery codes. The output false could mean that the otp is missing or **incorrect**. So using this method in `mfa_enabled_no_otp_provided?(user)` is misleading as an OTP could be present but incorrect.

Reply:
Would renaming to `mfa_enabled_invalid_otp_provided?` be clearer?